balancer-js/src/modules/zaps/migrations.ts

```diff
@@ -0,0 +1,175 @@
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { StaBal3Builder } from './bbausd2-migrations/stabal3';
+import { BbaUsd1Builder } from './bbausd2-migrations/bbausd1';
+import { StablesBuilder } from './bbausd2-migrations/stables';
+
+export class Migrations {
+  constructor(private network: 1 | 5) {}
+
+  /**
+   * Builds migration call data.
+   * Migrates tokens from staBal3 to bbausd2 pool.
+   * Tokens that are initially staked are re-staked at the end of migration. Non-staked are not.
+   *
+   * @param {string}  userAddress User address.
+   * @param {string}  staBal3Amount Amount of BPT tokens to migrate.
+   * @param {string}  minBbausd2Out Minimum of expected BPT out ot the migration flow.
+   * @param {boolean} staked Indicates whether tokens are initially staked or not.
+   * @param {string}  authorisation Encoded authorisation call.
+   * @returns Migration transaction request ready to send with signer.sendTransaction
+   */
+  stabal3(
+    userAddress: string,
+    staBal3Amount: string,
+    minBbausd2Out: string,
+    staked: boolean,
+    authorisation?: string
+  ): {
+    to: string;
+    data: string;
+    decode: (output: string, staked: boolean) => string;
+  } {
+    const builder = new StaBal3Builder(this.network);
+    const request = builder.calldata(
+      userAddress,
+      staBal3Amount,
+      minBbausd2Out,
+      staked,
+      authorisation
+    );
+
+    return {
+      to: request.to,
+      data: request.data,
+      decode: (output, staked) => {
```

Comment:
looks like it's not a pure decode anymore. Can we call it something that tells more about business logic? eg parseBptOut?